### PR TITLE
Tgp 2691/update download file notification banner

### PR DESCRIPTION
**Pull request by kenneth-rayner**
# Checklist for developers before reviewing
 - [x] I have merged in main
 - [x] I have assigned the PR to me
 - [x] I ran the code with a clean compile test and ensured there are no warnings or deprecations
 - [x] I have run `sbt prePR` to run all tests, format and check the code coverage
 - [x] I have written unit tests to cover any code that I wrote, where necessary
 - [x] I have linked tickets to any TODOs I have written
 - [x] I have run the microservices with the latest SM2 config
 - [x] I have checked all the Acceptance Criteria pass
 - [x] I have run the pipeline journey tests
 - [x] I have run the local journey tests
 - [x] I have spoken to the QAs about any changes that may need to be made to the journey tests OR no changes to the journey tests are necessary
 - [x] I have moved the ticket to Review in Jira
 - [x] I have left a message in tgp-prs on slack to alert other Devs that this ticket is ready for review
 - [x] I have checked my PR is not in draft mode

# Checklist for reviewer 1
- [x] I have added an emoji to the relevant message in tgp-prs to show I am reviewing
- [x] I have checked out the branch
- [x] I have run the microservices with the latest SM2 config
- [x] I have checked all the Acceptance Criteria pass
- [x] I reviewed the code and left comments where necessary
- [x] I have left a message on the relevant tgp-prs message to show that I have finished reviewing

# Checklist for reviewer 2
- [x] I have added an emoji to the relevant message in tgp-prs to show I am reviewing
- [ ] I have checked out the branch
- [ ] I have run the microservices with the latest SM2 config
- [ ] I have checked all the Acceptance Criteria pass
- [ ] I reviewed the code and left comments where necessary
- [ ] I have left a message on the relevant tgp-prs message to show that I have finished reviewing

# Checklist for developers before merging
 - [x] The PR has been approved by 2 other developers
 - [x] I have merged in main
 - [x] I have run `sbt preMerge` to run all tests, check formatting and code coverage
 - [x] I have run the microservices with the latest SM2 config
 - [x] I have checked all the Acceptance Criteria pass
 - [x] I have checked that the QAs have merged in any changes to the journey tests
 - [x] I have run the pipeline journey tests
 - [x] I have run the local journey tests

# Reminders for developers after merging
 1. Wait for pipeline to pass
 2. Move ticket to QA column in Jira
 3. Leave message in tgp-qa on slack to alert QAs that this ticket is ready for testing

### CODE REVIEW

**Review by yohan-oc — DISMISSED** [review]

File: app/views/HomePageView.scala.html [hunk 1]
```diff
@@ -14,25 +14,27 @@
...
 @this(
-        layout: templates.Layout,
-        govukNotificationBanner: GovukNotificationBanner
+layout: templates.Layout,
+govukPhaseBanner: GovukPhaseBanner,
+notificationBanner: notificationBanner,
+paragraphBody: paragraphBody
 )
 
-@(downloadReady: Boolean = false)(implicit request: Request[_], messages: Messages)
```
Comment (Reece-Carruthers) [inline app/views/HomePageView.scala.html:22]: Would we be able to add the default value back onto this? Unsure of the side effects of it being removed

File: app/views/HomePageView.scala.html [hunk 2]
```diff
@@ -14,25 +14,27 @@
...
-        @govukNotificationBanner(NotificationBanner(
-            title = Text(messages("successBanner.title")),
-            content = HtmlContent(f"""
-                <h3 class='govuk-notification-banner__heading'>
-                    ${messages("homepage.downloadBanner.content")}
-                    <a href="${routes.FileReadyController.onPageLoad().url}" class='govuk-link'>${messages("homepage.downloadBanner.linkText")}</a>.
-                </h3>"""),
-            bannerType = Some("success"),
-            role=Some("alert"))
+        @notificationBanner(
```
Comment (Reece-Carruthers) [inline app/views/HomePageView.scala.html]: Would we be able to still use the official govukNotificationBanner, but pass in the html, as this is the implementation on the component site
```
@import uk.gov.hmrc.govukfrontend.views.html.components._

@this(govukNotificationBanner : GovukNotificationBanner)

@()(implicit messages: Messages, request: RequestHeader)


@html = {
<p class="govuk-notification-banner__heading">
    You have 7 days left to send your application.
    <a class="govuk-notification-banner__link" href="#">View application</a>.
  </p>


}
@govukNotificationBanner(NotificationBanner(
    content = HtmlContent(html)
  ))
  
  ```
  You can have the new notificationBanner.scala.html be this part
  
  ```
  <p class="govuk-notification-banner__heading">
    You have 7 days left to send your application.
    <a class="govuk-notification-banner__link" href="#">View application</a>.
  </p>
  ```
  And take in a title and link to fill in the content

File: app/views/HomePageView.scala.html [hunk 3]
```diff
@@ -14,25 +14,27 @@
...
+paragraphBody: paragraphBody
 )
 
-@(downloadReady: Boolean = false)(implicit request: Request[_], messages: Messages)
+@(downloadReady: Boolean)(implicit request: Request[_], messages: Messages)
 
 @layout(pageTitle = s"${messages("homepage.title")} - ${messages("site.govuk")}", showBackLink = false, headerShouldHaveServiceLink = false) {
 
     @if(downloadReady) {
-        @govukNotificationBanner(NotificationBanner(
```
Comment (besscerule) [inline app/views/HomePageView.scala.html:27]: is there any reason why we are not just using the govukNotificationBanner?

File: app/views/HomePageView.scala.html (hunk 2 above)
Comment (kenneth-rayner) [reply app/views/HomePageView.scala.html]: Done.

File: app/views/HomePageView.scala.html (hunk 3 above)
Comment (kenneth-rayner) [reply app/views/HomePageView.scala.html:27]: @I have changed it back, I could not find the way to change the colour from green to blue as changes were not in plain sight so I used GovUk components. However I have now resolved this as per comments.

Comment (besscerule) [reply app/views/HomePageView.scala.html:27]: excellent - where possible always use govuk components as accessibility etc will be fully tested and they will be the accepted standard :)

**Review by besscerule — APPROVED** [review]

**Review by Reece-Carruthers — APPROVED** [review]

**Pull request merged**